shared/src/ledger/storage/write_log.rs

```diff
@@ -204,6 +241,17 @@ impl WriteLog {
             .collect()
     }
 
+    /// Get the IBC event of the current transaction
+    pub fn get_ibc_event(&self) -> Option<IbcEvent> {
```

Comment:
I think this can return `Option<&IbcEvent>` to avoid cloning (when you need an owned value for the `TxResult`, you can call `.cloned()` on it)